R/util_data_summaries.R

```diff
@@ -373,7 +373,7 @@ ds.glm1 <- function( formula, data, family=NULL) {
     # run the appropriate model, unless family override is supplied
     dv_variable <- ds.helper$glm1_formula_to_varlist(formula)[["dv"]]
     if (is.null(family)) {
-        dv_var_type <- ds.helper$variable_type(data[,dv_variable])
+        dv_var_type <- ds.helper$variable_type(data[[dv_variable]])
```

Comment:
This seems like it was definitely a bug before. Doesn't `[,dv_variable]` mean "subset this dataframe, taking all rows and the column named X" giving you a one-column df? If the goal is to extract the column as a vector, your new code is the way to go.

Reply:
The two notations are _usually_ equivalent, even though I agree that your interpretation makes more sense. I've never figured out the exact conditions under which the former notation "breaks" (i.e., exhibits behavior similar to what you describe—but instead of returning a one-column df, it returns a one-column matrix, which is just no good for anybody. I suspect it depends on which version of R you are running, though I could be wrong). But I've noticed that sometimes it does break, whereas the latter notation exhibits consistent behavior.